File: app/src/main/java/me/wcy/music/model/Music.java
package me.wcy.music.model;

import java.io.Serializable;

/**
 * 单曲信息
 * Created by wcy on 2015/11/27.
 */
public class Music implements Serializable {
    // 歌曲类型:本地/网络
    private Type type;
    // [本地歌曲]歌曲id
    private long id;
    // 音乐标题
    private String title;
    // 艺术家
    private String artist;
    // 专辑
    private String album;
    // [本地歌曲]专辑ID
    private long albumId;
    // [在线歌曲]专辑封面路径
    private String coverPath;
    // 持续时间
    private long duration;
    // 音乐路径
    private String path;
    // 文件名
    private String fileName;
    // 文件大小
    private long fileSize;

    public enum Type {
        LOCAL,
        ONLINE
    }

    public Type getType() {
        return type;
    }

    public void setType(Type type) {
        this.type = type;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getArtist() {
        return artist;
    }

    public void setArtist(String artist) {
        this.artist = artist;
    }

    public String getAlbum() {
        return album;
    }

    public void setAlbum(String album) {
        this.album = album;
    }

    public long getAlbumId() {
        return albumId;
    }

    public void setAlbumId(long albumId) {
        this.albumId = albumId;
    }

    public String getCoverPath() {
        return coverPath;
    }

    public void setCoverPath(String coverPath) {
        this.coverPath = coverPath;
    }

    public long getDuration() {
        return duration;
    }

    public void setDuration(long duration) {
        this.duration = duration;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public long getFileSize() {
        return fileSize;
    }

    public void setFileSize(long fileSize) {
        this.fileSize = fileSize;
    }

    /**
     * 对比本地歌曲是否相同
     */
    @Override
    public boolean equals(Object o) {
        return o instanceof Music && this.getId() == ((Music) o).getId();
    }
}

File: app/src/main/java/me/wcy/music/model/Music.kt
package me.wcy.music.model

import android.text.TextUtils
import java.io.Serializable

/**
 * 单曲信息
 * Created by wcy on 2015/11/27.
 */
class Music : Serializable {
    @JvmField
    var id: Long? = null

    @JvmField
    var type = 0 // 歌曲类型:本地/网络

    @JvmField
    var songId: Long = 0 // [本地]歌曲ID

    @JvmField
    var title: String? = null // 音乐标题

    @JvmField
    var artist: String? = null // 艺术家

    @JvmField
    var album: String? = null // 专辑

    @JvmField
    var albumId: Long = 0 // [本地]专辑ID

    @JvmField
    var coverPath: String? = null // [在线]专辑封面路径

    @JvmField
    var duration: Long = 0 // 持续时间

    @JvmField
    var path: String = "" // 播放地址

    @JvmField
    var fileName: String? = null // [本地]文件名

    @JvmField
    var fileSize: Long = 0 // [本地]文件大小

    interface Type {
        companion object {
            const val LOCAL = 0
            const val ONLINE = 1
        }
    }

    override fun equals(o: Any?): Boolean {
        if (o !is Music) {
            return false
        }
        val music = o
        if (music.songId > 0 && music.songId == songId) {
            return true
        }
        return (TextUtils.equals(music.title, title)
                && TextUtils.equals(music.artist, artist)
                && TextUtils.equals(music.album, album) && music.duration == duration)
    }

    companion object {
        private const val serialVersionUID: Long = 536871008
    }
}